SimPEG/directives.py

```diff
@@ -210,29 +230,27 @@ def initialize(self):
             :rtype: float
             :return: beta0
         """
+        if self.seed is not None:
+            np.random.seed(self.seed)
 
         if self.debug:
             print("Calculating the beta0 parameter.")
 
         m = self.invProb.model
         f = self.invProb.getFields(m, store=True, deleteWarmstart=False)
+        if len(self.dmisfit.objfcts) == 1:
+            f = [f]
 
-        # Fix the seed for random vector for consistent result
-        np.random.seed(1)
-        x0 = np.random.rand(*m.shape)
+        dm_eigenvalue = eigenvalue_by_power_iteration(
+            self.dmisfit, m, fields_list=f, n_pw_iter=self.n_pw_iter, 
+        )
+        reg_eigenvalue = eigenvalue_by_power_iteration(
+            self.reg, m, fields_list=f, n_pw_iter=self.n_pw_iter, 
```

Comment:
Also, I don't think you need to pass a `fields_list` here if I'm comprehending the code.